lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -733,7 +734,13 @@ def set_zsort(self, zsort):
             The function applied on the z-coordinates of the vertices in the
             viewer's coordinate system, to determine the z-order.
         """
-        self._zsortfunc = self._zsort_functions[zsort]
+        def nansafe(func):
```

Comment:
If this is seen as a sensible solution then I can add a docstring here saying why we do this. I could also add `functools.wraps` if desired?

Reply:
Are there many callers of `_zsort_func`? It's tempting just to insert this wrapping into the call site.

Reply:
There is only one call. I don't have a preference, so happy to change if you want.

Reply:
I think it probably makes sense to do it at the call site then

Reply:
The only mildly annoying thing about moving the wrapping to the call site
https://github.com/matplotlib/matplotlib/blob/498bcdda09058db2fd1aca9cc2437e5f2fa26055/lib/mpl_toolkits/mplot3d/art3d.py#L821-L827
is that it will push https://github.com/matplotlib/matplotlib/blob/498bcdda09058db2fd1aca9cc2437e5f2fa26055/lib/mpl_toolkits/mplot3d/art3d.py#L824
over the character limit. By no means the end of the world, but will involve adjusting more things than if we do it here. I can do it but my feeling is it will make an already slightly messy piece of code messier. I'm happy to change if you would still prefer the wrapping there?

Reply:
One more option would be to just make the wrapper a method, which avoids the need for the closures:
```python
        def _get_zsort_val(self, xs):
            value = self._zsortfunc(x)
            return np.inf if np.isnan(value) else value
```
then replace the one caller of `_zsort_func` with a call to `_get_zsort_val`